### PR TITLE
History / Revisions: Bump CoreData model version and new local models 

### PR DESCRIPTION
**Pull request by danielebogo**
Ref. #10304 
Part of #10358 

This PR fix the new entities structure (Revision and Diff) and introduces the new local models to store post revisions, revision diffs and diff values.

Data model changes:
- Renamed `RevisionDiffAbstractValue`, `RevisionDiffContentValue`, `RevisionDiffTitleValue` to `DiffAbstractValue`, `DiffContentValue`, `DiffTitleValue`.
- Set `DiffAbstractValue` as abstract entity which was omitted from model 82.
- Replaced relationship property name on `DiffContentValue` and `DiffTitleValue` from *relationship* to *revisionDiff*.
- Set `DiffAbstractValue` as parent entity of `DiffContentValue` which was omitted from model 82.
- Replaced properties name on `DiffAbstractValue` from *operation* to *diffOperation* and *type* to *diffType*.
- Added property *index* on `DiffAbstractValue` to store the right index position within the set.

Also this PR includes the new local models linked to the new entities.

**To test:**
- Run first develop
- Then run this PR on the top of the previous one to test the new datamodel version migration

### CODE REVIEW

File: MIGRATIONS.md [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
 data model as well as any custom migrations.
 
+## WordPress 83
+@danielebogo 2018-10-30
+- Renamed `RevisionDiffAbstractValue`, `RevisionDiffContentValue`, `RevisionDiffTitleValue` to `DiffAbstractValue`, `DiffContentValue`, `DiffTitleValue`.
+- Set `DiffAbstractValue` as abstract entity.
+- Replaced relationship property name on `DiffContentValue` and `DiffTitleValue` from *relationship* to *revisionDiff*.
+- Fix parent entity of `DiffContentValue`.
+- Replaced properties name on `DiffAbstractValue`.
+- Added property *index* on `DiffAbstractValue`.
```
Comment (aerych) [inline MIGRATIONS.md]: I'm thinking this means that we added an `index` attribute to the `DiffAbstractValue` model but I don't see it in the editor.  Have I misunderstood?
screen shot 2018-10-31 at 2 02 55 pm

File: WordPress/Classes/Models/Revisions/DiffAbstractValue.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    enum DiffType: String {
+        case title
+        case content
+        case unknown
+    }
+
+    @NSManaged private var diffOperation: String
+    @NSManaged private var diffType: String
+
+    @NSManaged var index: Int
```
Comment (aerych) [inline WordPress/Classes/Models/Revisions/DiffAbstractValue.swift]: Ah, yeah it looks like the new `index` property didn't make it into the model editor. Or maybe the intent wasn't for it to be managed?

Comment (danielebogo) [reply WordPress/Classes/Models/Revisions/DiffAbstractValue.swift]: 🤔It should be in the model editor. I can see it.
screenshot 2018-11-01 at 10 48 06

Comment (danielebogo) [reply WordPress/Classes/Models/Revisions/DiffAbstractValue.swift]: Also check this [commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/10368/commits/d07e8f3704a5993ad8c5ed2c109c14830da34ccb). You can see it as attribute of that model.

File: MIGRATIONS.md [hunk 3]
```diff
@@ -3,6 +3,15 @@
 This file documents changes in the data model. Please explain any changes to the
 data model as well as any custom migrations.
 
+## WordPress 83
+@danielebogo 2018-10-30
+- Renamed `RevisionDiffAbstractValue`, `RevisionDiffContentValue`, `RevisionDiffTitleValue` to `DiffAbstractValue`, `DiffContentValue`, `DiffTitleValue`.
+- Set `DiffAbstractValue` as abstract entity.
+- Replaced relationship property name on `DiffContentValue` and `DiffTitleValue` from *relationship* to *revisionDiff*.
+- Fix parent entity of `DiffContentValue`.
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply MIGRATIONS.md]: Done 👍

File: MIGRATIONS.md [hunk 4]
```diff
@@ -3,6 +3,15 @@
...
+- Replaced properties name on `DiffAbstractValue`.
```
[the remark this answers is not in the document]
Comment (danielebogo) [reply MIGRATIONS.md]: Done 👍

File: MIGRATIONS.md (hunk 1 above)
Comment (danielebogo) [reply MIGRATIONS.md]: I can see it on my model editor

File: WordPress/Classes/Models/Revisions/DiffAbstractValue.swift (hunk 2 above)
Comment (aerych) [reply WordPress/Classes/Models/Revisions/DiffAbstractValue.swift]: That's so weird. I refreshed the branch and no I see it too.
screen shot 2018-11-01 at 9 32 34 am

**Pull request merged**